src/import/FSHImporter.ts

```diff
@@ -205,6 +229,51 @@ export class FSHImporter extends FSHVisitor {
     });
   }
 
+  visitValueSet(ctx: pc.ValueSetContext) {
+    const valueSet = new ValueSet(ctx.SEQUENCE().getText())
+      .withLocation(this.extractStartStop(ctx))
+      .withFile(this.file);
+    this.parseValueSet(valueSet, ctx.vsMetadata(), ctx.vsComponent());
+    this.doc.valueSets.set(valueSet.name, valueSet);
+  }
+
+  private parseValueSet(
+    valueSet: ValueSet,
+    metaCtx: pc.VsMetadataContext[] = [],
+    componentCtx: pc.VsComponentContext[] = []
+  ) {
+    const seenPairs: Map<VsMetadataKey, string> = new Map();
+    metaCtx
+      .map(vsMetadata => ({
+        ...this.visitVsMetadata(vsMetadata),
+        context: vsMetadata
+      }))
+      .forEach(pair => {
+        if (seenPairs.has(pair.key)) {
+          logger.error(
```

Comment:
Ah, I forgot to add this because the duplicate metadata logic is the same as for other entities. But, they're different functions, so I'll add tests. Maybe I should also try to figure out how to not have that logic duplicated, too, given that the duplicate metadata rules applies to all entity types.